### PR TITLE
[OPPRO-190] Update micro benchmark to use Arrow C ABI

### PR DESCRIPTION
**Pull request by marin-ma**

### CODE REVIEW

File: cpp/src/jni/exec_backend.h [hunk 1]
```diff
@@ -241,7 +241,7 @@ class ArrowArrayResultIterator : public ResultIteratorBase<ArrowArray> {
   }
 
   /// ArrowArrayIterator doesn't support shared ownership. Once this method is
-  /// called, the caller should take it's ownership, and ArrowArrayResultIterator
+  /// called, the caller should take it's ownership, and ArrowExecResultIterator
```
Comment (jinchengchenghh) [inline cpp/src/jni/exec_backend.h:244]: Maybe we don't need to change here?

File: cpp/velox/compute/VeloxPlanConverter.cc [hunk 2]
```diff
@@ -254,6 +254,7 @@ std::shared_ptr<gluten::ArrowArrayResultIterator> VeloxPlanConverter::GetResultI
     arrowInputIters_ = std::move(inputs);
   }
   const std::shared_ptr<const core::PlanNode> planNode = getVeloxPlanNode(plan_);
+  cacheOutputSchema(planNode);
```
Comment (jinchengchenghh) [inline cpp/velox/compute/VeloxPlanConverter.cc:257]: I'm not sure if we could add extra code process just for benchmark?

Comment (marin-ma) [reply cpp/velox/compute/VeloxPlanConverter.cc:257]: Makes sense.

**Comment by jinchengchenghh** [comment]
I find you fix gazelle-cpp backend, can you add back arrow-backend github action?

**Comment by marin-ma** [comment]
@jinchengchenghh Ready to merge. Could you help approve this PR?

**Review by jinchengchenghh — APPROVED** [review]

**Pull request merged**